### PR TITLE
feat: nicer `z.function()` messages

### PR DESCRIPTION
**Pull request by abesto**
# Before

```
Validation error: Invalid function arguments
```

```
Validation error: Invalid function return type
```

# After

```
Validation error: Invalid function arguments; Expected number, received string at index 0
```

```
Validation error: Invalid function return type; Expected number, received string"
```

### CODE REVIEW

**Review by jmike — APPROVED** [review]
Yes, this is great!

**Review by thanoskrg — APPROVED** [review]
Thank you @abesto for taking care of this one 👍

File: lib/fromZodIssue.test.ts [hunk 1]
```diff
@@ -30,6 +30,84 @@ describe('fromZodIssue()', () => {
...
   });
 
+  test('handles zod.function() argument errors', () => {
+    const fn = zod
+      .function()
+      .args(zod.number())
+      .implement((num) => num * 2);
+
+    try {
+      // @ts-ignore
```
[the remark this answers is not in the document]
Comment (abesto) [reply lib/fromZodIssue.test.ts:40]: Right, usually `eslint` warns me about stuff like this so stopped thinking about it :D

[the remark this answers is not in the document]
Comment (thanoskrg) [reply lib/fromZodIssue.test.ts]: Valid point, an eslint rule would be a nice addition for that. Thanks updating it 👍 

cc @jmike

**Comment by jmike** [comment]
@abesto you should be able to merge this PR now, but let me know if not and I can push the button on your behalf.

**Comment by abesto** [comment]
@jmike please push the button :)

image

**Pull request merged**